pkg/controller/volume/persistentvolume/controller.go

```diff
@@ -1067,11 +1067,20 @@ func (ctrl *PersistentVolumeController) deleteVolumeOperation(arg interface{}) {
 	if err = ctrl.doDeleteVolume(volume); err != nil {
 		// Delete failed, update the volume and emit an event.
 		glog.V(3).Infof("deletion of volume %q failed: %v", volume.Name, err)
-		if _, err = ctrl.updateVolumePhaseWithEvent(volume, api.VolumeFailed, api.EventTypeWarning, "VolumeFailedDelete", err.Error()); err != nil {
-			glog.V(4).Infof("deleteVolumeOperation [%s]: failed to mark volume as failed: %v", volume.Name, err)
-			// Save failed, retry on the next deletion attempt
-			return
+		if vol.IsTryAgain(err) {
```

Comment:
Every controller sync, which is 15 seconds by default. And it does not start new delete until an old one (on the same PV) returns.